### PR TITLE
Google provider December removal

### PR DESCRIPTION
**Pull request by olegkachur-e**
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by moiseenkov — CHANGES_REQUESTED** [review]
Well done!
There are only two things I'd kindly ask you to amend:
1) please add an announcement about breaking changes and the major version bump in the `providers/src/airflow/providers/google/CHANGELOG.rst`  the same way it has been done [last time](https://github.com/apache/airflow/pull/43953).
2) please rename unit test classes, so the old operators' names were vanished from everywhere and the current tests had consistent names:
`TestCreateDataPipelineOperator` -> `TestDataflowCreatePipelineOperator`
`TestRunDataPipelineOperator` -> `TestDataflowRunPipelineOperator`

**Review by MaksYermak — APPROVED** [review]

**Comment by olegkachur-e** [comment]
> Well done! There are only two things I'd kindly ask you to amend:
> 
> 1. please add an announcement about breaking changes and the major version bump in the `providers/src/airflow/providers/google/CHANGELOG.rst`  the same way it has been done [last time](https://github.com/apache/airflow/pull/43953).
> 2. please rename unit test classes, so the old operators' names were vanished from everywhere and the current tests had consistent names:
>    `TestCreateDataPipelineOperator` -> `TestDataflowCreatePipelineOperator`
>    `TestRunDataPipelineOperator` -> `TestDataflowRunPipelineOperator`

Brilliant mentions, thank you!

**Comment by moiseenkov** [comment]
> > Well done! There are only two things I'd kindly ask you to amend:
> > 
> > 1. please add an announcement about breaking changes and the major version bump in the `providers/src/airflow/providers/google/CHANGELOG.rst`  the same way it has been done [last time](https://github.com/apache/airflow/pull/43953).
> > 2. please rename unit test classes, so the old operators' names were vanished from everywhere and the current tests had consistent names:
> >    `TestCreateDataPipelineOperator` -> `TestDataflowCreatePipelineOperator`
> >    `TestRunDataPipelineOperator` -> `TestDataflowRunPipelineOperator`
> 
> Brilliant mentions, thank you!

Thank you! All is good, but please move changelog announcements into a new section for the new 12.0.0 version.

File: providers/src/airflow/providers/google/CHANGELOG.rst [hunk 1]
```diff
@@ -27,6 +27,29 @@
...
+  The following breaking changes were introduced:
+
+  * Operators
+
+    * Removed ``CreateDataPipelineOperator``. Please use the ``DataflowCreatePipelineOperator`` instead
+    * Removed ``RunDataPipelineOperator``. Please use the ``DataflowRunPipelineOperator`` instead
+
+  * Hooks
+
+    * Removed ``DataPipelineHook``. Please use the DataflowHook instead
```
Comment (moiseenkov) [inline providers/src/airflow/providers/google/CHANGELOG.rst:47]:
```suggestion
    * Removed ``DataPipelineHook``. Please use the ``DataflowHook`` instead
```
nitpick

Comment (olegkachur-e) [reply providers/src/airflow/providers/google/CHANGELOG.rst]: Thank you! fixed.

**Review by moiseenkov — APPROVED** [review]

**Pull request closed without merging**